src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -1461,6 +1461,17 @@ impl<'a> TypeChecker<'a> {
                     data_type,
                 )))
             }
+            BinaryOperator::Like => {
+                // Convert `Like` to compare function , such as `p_type like PROMO%` will be converted to `p_type >= PROMO and p_type < PROMP`
+                if let Expr::Literal { lit, .. } = right {
+                    if let Literal::String(str) = lit {
+                        return self.resolve_like(op, span, left, right, str).await;
+                    }
+                }
```

Comment:
Maybe we can directly return error in else? Seems right must be String?

Reply:
It can be an expression as well.